common/lib/xmodule/xmodule/modulestore/mongo/draft.py

```diff
@@ -793,19 +793,26 @@ def delete_draft_only(root_location):
             versions_found = self.collection.find(
                 query, {'_id': True, 'definition.children': True}, sort=[SORT_REVISION_FAVOR_DRAFT]
             )
+            total_versions_found = versions_found.count()
+            published_version = [
+                version
+                for version in versions_found
+                if version.get('_id').get('revision') != MongoRevisionKey.draft
+            ][0]
```

Comment:
previously 0th versions_found was being used, comments explained that 0th would be publish and 1st would be draft, and on the next line it was being asserted that the versions_found[0] is not the draft in case it was fetched, so I just refactored that logic